packages/cloud-cognitive/src/components/InlineEdit/InlineEdit.test.js

```diff
@@ -75,4 +76,197 @@ describe(componentName, () => {
+        value={startingValue}
+        onChange={handleChange}
+        onSave={handleSave}
+      />
+    );
+    const component = container.querySelector(`.${blockClass}`);
+
+    const input = screen.getByRole('textbox');
+
+    // starts edit
+    const edit = container.querySelector(`.${blockClass}__edit`);
+    userEvent.click(edit);
+    expect(component).toHaveClass(`${blockClass}--editing`);
+
+    // types some text
+    userEvent.type(input, 'ABC');
+    expect(input).toHaveTextContent(`${startingValue}ABC`);
+
+    // has edit buttons
+    let controls = screen.getAllByRole('button');
+    expect(controls).toHaveLength(2); // cancel and save
+
+    // click cancel check result reverts
+    userEvent.click(controls[1]);
+    expect(input).toHaveTextContent(startingValue);
+    expect(component).not.toHaveClass(`${blockClass}--editing`);
+    expect(input).toHaveTextContent(`${startingValue}ABC`);
+
+    expect(handleChange).toBeCalledTimes(3);
+    expect(handleSave).toBeCalled();
+  });
+
+  it('does not allow disabled edit', async () => {
+    const startingValue = value;
+
+    const { container } = render(
+      <InlineEdit {...requiredProps} value={startingValue} disabled={true} />
+    );
+    const component = container.querySelector(`.${blockClass}`);
+
+    // starts edit
+    const edit = container.querySelector(`.${blockClass}__edit`);
+    userEvent.click(edit);
+    expect(component).not.toHaveClass(`${blockClass}--editing`);
+  });
+
+  it('Clicking control but not on edit or input starts edit', async () => {
+    const startingValue = value;
+
+    const { container } = render(
+      <InlineEdit {...requiredProps} value={startingValue} />
+    );
+    const component = container.querySelector(`.${blockClass}`);
+
+    // starts edit clicking on background (left of input) as if clicking left edge of input
+    userEvent.click(component);
+    expect(component).toHaveClass(`${blockClass}--editing`);
+
+    // tab out cancelling edit
+
+    userEvent.click(screen.getAllByRole('button')[0]);
+    expect(component).not.toHaveClass(`${blockClass}--editing`);
+
+    // starts edit clicking in toolbar area but not on edit as if clicking right edge of input
+    const afterInput = container.querySelector(
+      `.${blockClass}__after-input-elements`
+    );
+    userEvent.click(afterInput);
+    expect(component).toHaveClass(`${blockClass}--editing`);
+  });
+
+  it.skip('Handles clipboard data', async () => {
+    const startingValue = value;
+    const clipboardString = 'AString';
+
+    render(<InlineEdit {...requiredProps} value={startingValue} />);
+
+    const input = screen.getByRole('textbox');
+    userEvent.click(input);
+    userEvent.paste(input, clipboardString);
+
+    // paste event not supported with div as yet
+    // https://github.com/testing-library/user-event/pull/785
+    expect(input).toHaveTextContent(startingValue + clipboardString + 'b');
+  });
+
+  it('Can show warn state', async () => {
```

Comment:
You don't need the async as you're not doing any awaits.

```suggestion
  it('Can show warn state', () => {
```